### PR TITLE
Implement bentley server based licensing for studio 1377303

### PR DESCRIPTION
**Pull request by ChenLeoB**

### CODE REVIEW

File: install [hunk 1]
```diff
@@ -107,6 +119,12 @@ function parseCommandLineArguments() {
                   --default-apache-app  default apache app to redirect to (defaults to projectmanager)
                   --sendgrid-api-key)   sendgrid apikey that will be used to send emails when there is an error in a cron backup job
                   --alert-email)        email address to send alerts to when a backup job fails
+                  --licensedir
```
Comment (CharlieASpida) [inline install]: please add info for these options

**Review by CharlieASpida — APPROVED** [review]

File: install [hunk 2]
```diff
@@ -107,6 +119,12 @@ function parseCommandLineArguments() {
                   --default-apache-app  default apache app to redirect to (defaults to projectmanager)
                   --sendgrid-api-key)   sendgrid apikey that will be used to send emails when there is an error in a cron backup job
                   --alert-email)        email address to send alerts to when a backup job fails
+                  --licensedir          directory for license executable and x-platform directory to initialize license checker
+                  --licenseenvironment  assigned environment type of the server for license purposes
```
Comment (spidaJeffSeifert) [inline install]: there are a few valid values - can those be given in the comments

Comment (ChenLeoB) [reply install]: do you mean the QA, RELEASE, DEV options?

**Review by SPIDABrianB — APPROVED** [review]

File: install [hunk 3]
```diff
@@ -308,7 +308,7 @@ function createDockerComposeFile() {
     if [[ -n "$licenseAccessKey" ]]; then
       echo "LICENSE_ACCESS_KEY=${licenseAccessKey}" >> $dockerEnvFile
     fi
-    if [[ -n "$licenseOfflineMode" ]]; then
+    if [[ $licenseOfflineMode = true ]]; then
```
Comment (CharlieASpida) [inline install:311]: make line 48
licenseOfflineMode=false
and line 92
--licenseofflinemode) licenseOfflineMode=true;;

**Pull request closed without merging**